triggers/inotify/inotify.go

```diff
@@ -0,0 +1,308 @@
+package inotify
+
+import (
+	"fmt"
+	"github.com/cloudbox/autoscan"
+	"github.com/fsnotify/fsnotify"
+	"github.com/rs/zerolog"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+)
+
+type Config struct {
+	Priority  int                `yaml:"priority"`
+	Verbosity string             `yaml:"verbosity"`
+	Rewrite   []autoscan.Rewrite `yaml:"rewrite"`
+	Include   []string           `yaml:"include"`
+	Exclude   []string           `yaml:"exclude"`
+	Paths     []struct {
+		Path    string             `yaml:"path"`
+		Rewrite []autoscan.Rewrite `yaml:"rewrite"`
+		Include []string           `yaml:"include"`
+		Exclude []string           `yaml:"exclude"`
+	} `yaml:"paths"`
+}
+
+type daemon struct {
+	callback autoscan.ProcessorFunc
+	paths    []path
+	watcher  *fsnotify.Watcher
+	queue    *queue
+	log      zerolog.Logger
+}
+
+type path struct {
+	Path     string
+	Rewriter autoscan.Rewriter
+	Allowed  autoscan.Filterer
+}
+
+func New(c Config) (autoscan.Trigger, error) {
+	l := autoscan.GetLogger(c.Verbosity).With().
+		Str("trigger", "inotify").
+		Logger()
+
+	var paths []path
+	for _, p := range c.Paths {
+		p := p
+
+		rewriter, err := autoscan.NewRewriter(append(p.Rewrite, c.Rewrite...))
+		if err != nil {
+			return nil, err
+		}
+
+		filterer, err := autoscan.NewFilterer(append(p.Include, c.Include...), append(p.Exclude, c.Exclude...))
+		if err != nil {
+			return nil, err
+		}
+
+		paths = append(paths, path{
+			Path:     p.Path,
+			Rewriter: rewriter,
+			Allowed:  filterer,
+		})
+	}
+
+	trigger := func(callback autoscan.ProcessorFunc) {
+		d := daemon{
+			log:      l,
+			callback: callback,
+			paths:    paths,
+			queue:    newQueue(callback, l, c.Priority),
+		}
+
+		// start job(s)
+		if err := d.StartMonitoring(); err != nil {
+			l.Error().
+				Err(err).
+				Msg("Failed initialising jobs")
+			return
+		}
+	}
+
+	return trigger, nil
+}
+
+func (d *daemon) StartMonitoring() error {
```

Comment:
should this function be exported?

Reply:
updated this and the equivelant bernard one.